torchrl/objectives/decision_transformer.py

```diff
@@ -0,0 +1,320 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import math
+from dataclasses import dataclass
+from typing import Union
+
+import numpy as np
+
+import torch
+from tensordict.nn import dispatch
+from tensordict.tensordict import TensorDict, TensorDictBase
+from tensordict.utils import NestedKey
+
+from torch import distributions as d
+from torchrl.modules import ProbabilisticActor
+
+from torchrl.objectives.common import LossModule
+from torchrl.objectives.utils import distance_loss
+
+
+class OnlineDTLoss(LossModule):
+    r"""TorchRL implementation of the Online Decision Transformer loss.
+
+    Presented in `"Online Decision Transformer" <https://arxiv.org/abs/2202.05607>`
+
+    Args:
+        actor_network (ProbabilisticActor): stochastic actor
+        alpha_init (float, optional): initial entropy multiplier.
+            Default is 1.0.
+        min_alpha (float, optional): min value of alpha.
+            Default is None (no minimum value).
+        max_alpha (float, optional): max value of alpha.
+            Default is None (no maximum value).
+        fixed_alpha (bool, optional): if ``True``, alpha will be fixed to its
+            initial value. Otherwise, alpha will be optimized to
+            match the 'target_entropy' value.
+            Default is ``False``.
+        target_entropy (float or str, optional): Target entropy for the
+            stochastic policy. Default is "auto", where target entropy is
+            computed as :obj:`-prod(n_actions)`.
+        samples_mc_entropy (int): number of samples to estimate the entropy
```

Comment:
```suggestion
    Args:
        actor_network (ProbabilisticActor): stochastic actor

    Keyword Args:
        alpha_init (float, optional): initial entropy multiplier.
            Default is 1.0.
        min_alpha (float, optional): min value of alpha.
            Default is None (no minimum value).
        max_alpha (float, optional): max value of alpha.
            Default is None (no maximum value).
        fixed_alpha (bool, optional): if ``True``, alpha will be fixed to its
            initial value. Otherwise, alpha will be optimized to
            match the 'target_entropy' value.
            Default is ``False``.
        target_entropy (float or str, optional): Target entropy for the
            stochastic policy. Default is "auto", where target entropy is
            computed as :obj:`-prod(n_actions)`.
        samples_mc_entropy (int): number of samples to estimate the entropy.
            Default is 1.

```